lib/p2p/Pool.ts

```diff
@@ -312,6 +312,18 @@ class Pool extends EventEmitter {
     }
   }
 
+  private handleGetNodes = (peer: Peer, reqId: string) => {
+    const nodesInfo = this.nodes.toConnectionInfoArray();
+    const connectedNodesInfo: NodeConnectionInfo[] = [];
+    nodesInfo.forEach((nodeInfo) => {
```

Comment:
`toConnectionInfoArray` is already iterating over all nodes. This way we iterate over the nodes twice. Is it feasible to check if the node isn't banned, has addresses and is connected in one step?

Reply:
Thanks, I think I took care of this with the new approach.